### DISCUSSION TITLE
SSH with Public Key

### DISCUSSION THREAD

**Discussion by agluck91** [blinksh/blink#1757] [Q&A] [answered]
This one is driving me nuts. I can not figure this out. 

I am trying to enable public key auth on ubuntu server 22.05

I have public key auth enabled on my server
I have all the premissions right. 

In blink I have no password set, and I have my key added. 

Every time I try to login it is still asking for my password. There is not even an attempt to login with the key in the logs which is why I am coming here now. What in the world am I doing wrong here??? 

Thanks for the help in advance

**Comment by carloscabanero** [comment]
Can you include a `ssh -vvv` output?

**Reply by agluck91** [reply]
Here is the last bit, looks like it tried public key. 

```
Starting connection to 10.10.1.76
ssh_connect: current state : 7
Connection succeeded...
Authenticating...
Trying none...
packet_send2: packet: wrote [type=5, len=32, padding_size=14, comp=17, payload=17]
ssh_service_request: Sent SSH_MSG_SERVICE_REQUEST (service ssh-userauth)
ssh_packet_socket_callback: packet: read type 6 [len=32,padding=14,comp=17,payload=17]
ssh_packet_process: Dispatching handler for packet type 6
ssh_packet_service_accept: Received SSH_MSG_SERVICE_ACCEPT
packet_send2: packet: wrote [type=50, len=48, padding_size=10, comp=37, payload=37]
ssh_packet_socket_callback: packet: read type 51 [len=32,padding=7,comp=24,payload=24]
ssh_packet_process: Dispatching handler for packet type 51
ssh_packet_userauth_failure: Access denied for 'none'. Authentication that can continue: publickey,password
ssh_packet_userauth_failure: Access denied for 'none'. Authentication that can continue: publickey,password
Trying publickey...
agent_talk: Request length: 1
ssh_agent_get_ident_count: Answer type: 12, expected answer: 12
ssh_agent_get_ident_count: Agent count: 0
Trying password-interactive...
```

**Reply by carloscabanero** [reply]
Ok, so I think the key is not offered to the server. I was expecting something like:
`ssh_userauth_agent: Public key of id_ecdsa refused by server`

Are you using a standard name for the key, or did you select the key in the host configuration?

**Reply by agluck91** [reply]
Ummm, I create the key in host and keys if that is what you mean, and I used ED25519

**Reply by agluck91** [reply]
I used host and keys option under config in blink

**Reply by carloscabanero** [reply]
Yes, but then the key can have a default name (id_ed25519 if default) or custom. If custom, then the “key” section inside Host should have it selected. Note “custom” is the name I gave the key, it could be anything else. It should look like this:
![image](https://user-images.githubusercontent.com/536729/233243156-816ef587-d08c-4c54-bdc2-5ec5f12fbb54.png)

**Reply by agluck91** [reply]
ooooh, it is my custom key that I made. it looks like that.

**Reply by agluck91** [reply]
<img width="722" alt="image" src="https://user-images.githubusercontent.com/113385018/233246305-bc0c77c8-5268-41f4-8a6a-f008e92ab5cf.png">

**Reply by carloscabanero** [reply]
Ok, so I am sure the issue is that there is no key being offered to the server. Are you getting now the “public key of xxxx refused by server”? Maybe there is a permission problem with the authorized_keys file (https://loneidealist.medium.com/enabling-public-key-ssh-authentication-on-your-vps-fe5bfbf94822)

Plan B would be to take a look at what the server is saying. That would be on /var/log/auth.log on Linux or system.log on macOS.

**Reply by agluck91** [reply]
I have set the permissions to what the docs said. I think is was like 755 for .ssh folder and like 600 on the key file. I have enable public key auth in sshd. 

On the server in the logs there is not even an attempt to login with the key. The only thing in the log is my successful login with password. So I don’t think blink ssh is sending the key over to even try the auth.

**Reply by agluck91** [reply]
Let me try from my phone really quick

**Reply by agluck91** [reply, marked as answer]
Hmmm works with my phone. Maybe I just need to recreate my key.

**Reply by carloscabanero** [reply]
Did that work? Sorry I abandoned you when I went to sleep.

**Reply by agluck91** [reply]
My bad, yes I remade the key on my phone and imported it to my iPad and that did work. Not sure what changed in all that, but at least it works!
